Implement type support in RDGSlice

scober opened the pull request:
Mirror the type support in RDG, but load only a slice of the type id arrays, just like with the topology array in RDGSlice.

Follows #403

insertinterestingnamehere commented on conda_recipe/meta.yaml:
```diff
@@ -1,5 +1,15 @@
+{% set ci_build = environ.get("CI", False) %}
+{% set dev_build = environ.get("GALOIS_DEV_BUILD", ci_build) %}
+
+# For dev builds, add ".devN" to the version number, for N being the git commit count since the last tag.
```
Interesting idea. In the past when doing this kind of thing we just set an alpha version number and incremented the build number. This seems like it'll work similarly well.

arthurp replied:
I think it should work the same, but it makes the version number more visible. Either way would have worked.

Commit:
Implement type support in RDGSlice

Mirror the type support in RDG, but load only a slice of the type id
arrays, just like with the topology array in RDGSlice.

amberhassaan reviewed (approved):
looks good.

tylershunt reviewed (approved):
thanks
